### ISSUE TITLE
ko.oberservableArray length attribute is not correct.

### ISSUE THREAD

**Issue by digi604** [knockout/knockout#2035]
a = ko.obersevableArray([1])
a.length == 0: true
a.length == 1:false
a().length == 1: true

**Comment by brianmhunt** [comment]
Thanks @digi604 – Good idea.

ES3 did not support overwriting the `length` property, but as of ES5 we can get this in there, I do believe.

**Comment by brianmhunt** [comment]
Here's a great post that covers the options (among a great many other details): http://perfectionkills.com/how-ecmascript-5-still-does-not-allow-to-subclass-an-array/

**Comment by sateffen** [comment]
No, you can't overwrite the length property for the *observableArray*.

The *ko.observableArray* is a function, see [here](https://github.com/knockout/knockout/blob/master/src/subscribables/observableArray.js) line 1. The length of a function is defined as the number named parameters it can take.

The length property of a function is not writable, see [here](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Function/length), so sadly this can't be done. But you could add an property called *arrayLength* or a method called *getLength()*.

**Comment by brianmhunt** [comment]
Thanks @sateffen –

The link you point to shows that as of ES6 the `.length` is configurable.  So in Chrome 37+ and other ES6-compliant browsers the following works:

```javascript
>>> function foozball() {}
>>> Object.defineProperty(foozball, 'length', { get: () => 12 })
>>> foozball.length
12
```

It's slightly more complicated but it looks like essentially the same can be accomplished with `ko.observableArray` instances.

I doubt the majority of browsers-by-use support this right now, though, but in the ES6 world it is doable (unless I'm missing something :)).  Albeit with probably quite spotty reliability that cannot be polyfilled/shimmed.

Cheers

**Comment by sateffen** [comment]
You're right, I totally oversaw this.

But here the problem is, internet explorer I guess. IE11 is used in a lot of cases, but isn't able to handle any ES6 as far as I know. And dropping support for IE down to version 10.. Don't know. I'd still prefer the way of using an own method for this over overwriting something default.
